### PR TITLE
Nc news 5

### PR DESCRIPTION
**Pull request by Harrywg**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -50,6 +50,31 @@ describe("/api/topics", () => {
...
 
 describe("/api/articles", () => {
+  describe("GET", () => {
+    test("200 + returns all articles", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          const articles = body.articles;
+          articles.forEach((article) => {
```
Comment (johnhutcheon) [inline __tests__/app.test.js:62]: As you're asserting within a forEach, could you please check the length of the array beforehand? 👍

File: __tests__/app.test.js [hunk 2]
```diff
@@ -50,6 +50,31 @@ describe("/api/topics", () => {
...
+              expect.objectContaining({
+                article_id: expect.any(Number),
+                title: expect.any(String),
+                topic: expect.any(String),
+                author: expect.any(String),
+                body: expect.any(String),
+                created_at: expect.any(String),
+                votes: expect.any(Number),
+                article_img_url: expect.any(String),
+              })
```
Comment (johnhutcheon) [inline __tests__/app.test.js:73]: The ticket says that we don't want body to come back, whilst we also want to create 'comment_count'. Take a look at my comment in your model for more info 🙂

File: __tests__/app.test.js [hunk 3]
```diff
@@ -50,6 +50,31 @@ describe("/api/topics", () => {
...
+            );
+          });
+        });
+    });
+  });
+
```
Comment (johnhutcheon) [inline __tests__/app.test.js:79]: Either within this test, or in a new one, we also want to be checking the default sort and order please

File: models/articles-model.js [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-      `SELECT * FROM articles 
-      WHERE article_id = $1`,
-      [id]
-    )
-    .then(({ rows }) => {
-      return rows;
-    });
+exports.selectArticles = (params) => {
+  let query = `SELECT * FROM articles`;
+
```
Comment (johnhutcheon) [inline models/articles-model.js:5]: Your model needs a bit of work here. The ticket says that you should be fetching back comment_count, which is the total count of all the comments for each article_id. Take a look at the notes at the link below, which will help you out with this. 👍 
https://notes.northcoders.com/courses/js-back-end/sql-relationships#group-by-and-aggregate-functions

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hey Harry, nice work so far! Can you just see my comments please and make the changes, then resubmit? Thanks!

File: __tests__/app.test.js [hunk 5]
```diff
@@ -50,6 +50,60 @@ describe("/api/topics", () => {
...
+                comment_count: expect.any(String),
+                created_at: expect.any(String),
+                votes: expect.any(Number),
+                article_img_url: expect.any(String),
+              })
+            );
+          });
+        });
+    });
+    test("200 + returns all articles sorted by created_at", () => {
```
Comment (MRombi) [inline __tests__/app.test.js]: We could delete this test block and add expect(articles).toBeSortedBy("created_at", { descending: true }); to the previous one, it should achieve the same result 🙂

**Review by MRombi — APPROVED** [review]
Well done on this ticket, please follow my feedback and merge, no need to resubmit 😄

**Pull request merged**